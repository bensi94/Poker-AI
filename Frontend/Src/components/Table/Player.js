import React from 'react';
import PropTypes from 'prop-types';

import Stack from './Stack';

class Player extends React.Component {
    constructor(props) {
        super(props);
        let stateObj = {};

        switch (this.props.seatNumber) {
            case 1:
                stateObj = {
                    rectLeft: '70%',
                    rectTop: '-4%',
                    stackLeft: 80,
                    stackTop: 13
                };
                break;
            case 2:
                stateObj = {
                    rectLeft: '86%',
                    rectTop: '18%',
                    stackLeft: 90,
                    stackTop: 36
                };
                break;
            case 3:
                stateObj = {
                    rectLeft: '90%',
                    rectTop: '57%',
                    stackLeft: 86,
                    stackTop: 72
                };
                break;
            case 4:
                stateObj = {
                    rectLeft: '68%',
                    rectTop: '87%',
                    stackLeft: 71,
                    stackTop: 82
                };
                break;
            case 5:
                stateObj = {
                    rectLeft: '42%',
                    rectTop: '87%',
                    stackLeft: 50,
                    stackTop: 82
                };
                break;
            case 6:
                stateObj = {
                    rectLeft: '16%',
                    rectTop: '87%',
                    stackLeft: 30,
                    stackTop: 82
                };
                break;
            case 7:
                stateObj = {
                    rectLeft: '-6%',
                    rectTop: '57%',
                    stackLeft: 18,
                    stackTop: 72
                };
                break;
            case 8:
                stateObj = {
                    rectLeft: '-2%',
                    rectTop: '18%',
                    stackLeft: 13,
                    stackTop: 36
                };
                break;
            case 9:
                stateObj = {
                    rectLeft: '14%',
                    rectTop: '-4%',
                    stackLeft: 23.5,
                    stackTop: 13
                };
            default:
                break;
        }

        this.state = {
            ...stateObj,
            widthPercent: '16%'
        };
    }


    render() {
        return (
            <>
                <div style={{ width: this.state.widthPercent, left: this.state.rectLeft, top: this.state.rectTop }}
                    className="player-rect"
                >
                    <svg id = "92f50c61-a635-413f-87df-d274975daa0e" viewBox="0 0 150 75">
                        <title>player-rect</title>
                        <g id="5f2ad952-a9de-49e6-aadd-7f71fef5b04b" data-name="Seat 1">
                            <rect width="150" height="75" rx="5" ry="5" className="player-rect-bg"/>
                            <text transform="translate(75 56.62)"className="player-react-stack-text">
                                ${ this.props.stack }
                            </text>
                            <text transform="translate(75 21.96)" className="player-react-title-text">
                                { this.props.name }
                            </text>
                        </g>
                    </svg>
                </div>
                {this.props.bet > 0 ?
                    <Stack
                        tableSize={this.props.tableSize}
                        stackAmount={this.props.bet}
                        left={this.state.stackLeft}
                        top={this.state.stackTop}
                    /> : undefined
                }
            </>
        );
    }
}

Player.defaultProps = {
    bet: 0
};

Player.propTypes = {
    tableSize: PropTypes.number.isRequired,
    stack: PropTypes.number.isRequired,
    bet: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    seatNumber: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9])

};

export default Player;
