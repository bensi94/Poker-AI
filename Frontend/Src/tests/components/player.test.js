import React from 'react';
import { shallow } from 'enzyme';

import Player from '../../components/Table/Player';


describe('Player test sutie', () => {
    let wrapper;

    it('should render component correctly', () => {
        wrapper = shallow(
            <Player tableSize={1000}
                stack={100}
                bet={10}
                name="bensi"
                seatNumber={1}
                status="ACTIVE"
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should set the correct state for given seatNumber', () => {
        wrapper = shallow(
            <Player tableSize={1000}
                stack={100}
                bet={10}
                name="bensi"
                seatNumber={1}
                cards={['Kc', 'Kh']}
                status="ACTIVE"
            />
        );
        let caseStateObj = {
            rectLeft: '70%',
            rectTop: '-14%',
            stackLeft: 80,
            stackTop: 13,
            widthPercent: '16%',
            cardOpacity: {
                opacity: 1
            }
        };
        expect(wrapper.instance().state).toStrictEqual(caseStateObj);
    });

    it('should have a visual stack if there is bet', () => {
        wrapper = shallow(
            <Player tableSize={1000}
                stack={100}
                bet={10}
                name="bensi"
                seatNumber={1}
                cards={['Kc', 'Kh']}
                status="ACTIVE"
            />
        );
        expect(wrapper.find('Stack').length).toBe(1);
    });

    it('should not have a visual stack if there is no bet', () => {
        wrapper = shallow(
            <Player tableSize={1000}
                stack={100}
                name="bensi"
                seatNumber={1}
                status="FOLDED"
            />
        );
        expect(wrapper.find('Stack').length).toBe(0);
    });
});
