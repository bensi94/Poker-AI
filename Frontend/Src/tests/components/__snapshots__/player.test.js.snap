// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player test sutie should render component correctly 1`] = `
<Fragment>
  <div
    className="player-rect"
    style={
      Object {
        "left": "70%",
        "top": "-14%",
        "width": "16%",
      }
    }
  >
    <div
      className="player-card-wrapper"
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <div
        className="card-wrapper player-card-back"
        key="left"
      >
        <stubbed-asset />
      </div>
      <div
        className="card-wrapper"
        key="right"
      >
        <stubbed-asset />
      </div>
    </div>
    <svg
      id="92f50c61-a635-413f-87df-d274975daa0e"
      viewBox="0 0 150 75"
    >
      <title>
        player-rect
      </title>
      <g
        data-name="Seat 1"
        id="5f2ad952-a9de-49e6-aadd-7f71fef5b04b"
      >
        <rect
          className="player-rect-bg"
          height="75"
          rx="5"
          ry="5"
          width="150"
        />
        <text
          className="player-react-stack-text"
          transform="translate(75 56.62)"
        >
          $
          100
        </text>
        <text
          className="player-react-title-text"
          transform="translate(75 21.96)"
        >
          bensi
        </text>
      </g>
    </svg>
  </div>
  <Stack
    left={80}
    stackAmount={10}
    tableSize={1000}
    top={13}
  />
</Fragment>
`;
